### PR TITLE
user branch

### PR DESCRIPTION
**Pull request by Code-250**

### CODE REVIEW

File: Src/App/Controllers/PostsController.js [hunk 1]
```diff
@@ -111,13 +103,12 @@ export const updatePost = async (request, response) => {
     if (updateData) {
       response.status(200).json({
         status: 200,
-        message: `Post withid ${id} updatedSuccessfully`,
-        data: updateData,
+        message: `Post withid ${id} updated Successfully`,
```
Comment (RedJanvier) [inline Src/App/Controllers/PostsController.js:106]: why no data is being sent now?

File: Src/App/Controllers/authController.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import { secretKey } from "../../Database/Services/system/authConfig.js";
+import jwt from "jsonwebtoken";
+import bcrypt from "bcrypt";
+import Post from "../../Database/models/PostsModel.js";
+
+const Op = User.Sequelize.Op;
+export const signUp = async (request, response) => {
+  // save User to database
+
+  User.create({
```
Comment (RedJanvier) [inline Src/App/Controllers/authController.js:11]: isn't this supposed to return a promise??? Why aren't you awaiting it??

File: Src/App/Controllers/userController.js [hunk 3]
```diff
@@ -0,0 +1,272 @@
+import Sequelize from "sequelize";
+import bcrypt from "bcryptjs";
+import jwt from "jsonwebtoken";
+<<<<<<< HEAD
+import User from "../../Database/models/UserModels.js";
+=======
+<<<<<<< HEAD
+<<<<<<< HEAD
+import User from "../../Database/models/UserModels.js";
+import { response } from "express";
```
Comment (RedJanvier) [inline Src/App/Controllers/userController.js]: do not import things your are not using!

File: package.json [hunk 4]
```diff
@@ -5,19 +5,24 @@
   "main": "src/server.js",
   "type": "module",
   "dependencies": {
+    "@hapi/joi": "^17.1.1",
```
Comment (RedJanvier) [inline package.json:8]: you need to use Joi instead of @hapi/joi.
Don't use deprecated packages.

File: package.json [hunk 5]
```diff
@@ -5,19 +5,24 @@
...
     "babel": "^6.23.0",
     "bcrypt": "^5.0.0",
```
Comment (RedJanvier) [inline package.json:10]: you must remove bcrypt if you are using bcryptjs now!

File: nodemon.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
+{
+  "env":{
+    "JWT_KEY":"richardsecret"
```
Comment (RedJanvier) [inline nodemon.json:3]: You should move this to `.env` and please don't push `.env`

File: Src/Database/Services/system/db_config.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+export const HOST = "localhost";
```
Comment (RedJanvier) [inline Src/Database/Services/system/db_config.js]: This file is not a service.
you need to move it to the database level.

File: Src/Database/Services/secretKey.js [hunk 8]
```diff
@@ -0,0 +1,3 @@
+export const secret = {
+  env: "richardsecret",
```
Comment (RedJanvier) [inline Src/Database/Services/secretKey.js:2]: This is not a service!!!
You must also move all your secrets into `.env`...
DON'T PUSH YOUR SECRETS TO GITHUB!!!

File: .env [hunk 9]
```diff
@@ -1 +1,3 @@
-PORT = 8080
\ No newline at end of file
+
```
Comment (RedJanvier) [inline .env]: Please don't push this file.

**Review by RedJanvier — CHANGES_REQUESTED** [review]
@Code-250 
Please go through your conflicts and fix them. The PR is full of unresolved conflicts...
Make sure to keep your secrets locally. Don't push them to github.
And lastly don't merge develop into this branch!! You were supposed to rebase not merge!

File: Src/App/Controllers/PostsController.js (hunk 1 above)
Comment (Code-250) [reply Src/App/Controllers/PostsController.js:106]: data n isavailable

File: Src/Database/Services/system/db_config.js (hunk 7 above)
Comment (Code-250) [reply Src/Database/Services/system/db_config.js]: done it is in system folder

**Pull request merged**